src/tools/blast/blastcommandline.jl

```diff
@@ -0,0 +1,170 @@
+"""
+function blastn(query::AbstractString, subject::AbstractString, flags=[]; db::Bool=false)
+    if db
+        results = readblastXML(`blastn -query $query -db $subject $flags -outfmt 5`)
+    else
+        results = readblastXML(`blastn -query $query -subject $subject $flags -outfmt 5`)
+    end
+    return results
+end
+
+function blastn(query::BioSequence{DNAAlphabet{4}}, subject::BioSequence{DNAAlphabet{4}}, flags=[])
+    querypath, subjectpath = makefasta(query), makefasta(subject)
+    blastn(querypath, subjectpath, flags)
+end
+
+function blastn{S <: BioSequence{DNAAlphabet{4}}}(query::BioSequence{DNAAlphabet{4}}, subject::Vector{S}, flags=[])
+    querypath, subjectpath = makefasta(query), makefasta(subject)
+    blastn(querypath, subjectpath, flags)
+end
+
+function blastn(query::BioSequence{DNAAlphabet{4}}, subject::AbstractString, flags=[]; db::Bool=false)
+    querypath = makefasta(query)
+    if db
+        blastn(querypath, subject, flags, db=true)
+    else
+        blastn(querypath, subject, flags)
+    end
+end
+
+function blastn{S <: BioSequence{DNAAlphabet{4}}}(query::Vector{S}, subject::Vector{S}, flags=[])
+    querypath, subjectpath = makefasta(query), makefasta(subject)
+    blastn(querypath, subjectpath, flags)
+end
+
+"""
+`blastp(query, subject, flags...)``
+Runs blastn on `query` against `subject`.
+    Subjects and queries may be file names (as strings), `BioSequence{AminoAcidSequence}` type or
+    Array of `BioSequence{AminoAcidSequence}`.
+    May include optional `flag`s such as `["-perc_identity", 95,]`. Do not use `-outfmt`.
+"""
+function blastp(query::AbstractString, subject::AbstractString, flags=[]; db::Bool=false)
+    if db
+        results = readblastXML(`blastp -query $query -db $subject $flags -outfmt 5`, seqtype = "prot")
+    else
+        results = readblastXML(`blastp -query $query -subject $subject $flags -outfmt 5`, seqtype = "prot")
+    end
+end
+
+function blastp(query::AminoAcidSequence, subject::AminoAcidSequence, flags=[])
+    querypath, subjectpath = makefasta(query), makefasta(subject)
+    blastp(querypath, subjectpath, flags)
+end
+
+function blastp{S <: AminoAcidSequence}(query::AminoAcidSequence, subject::Vector{S}, flags=[])
+    querypath, subjectpath = makefasta(query), makefasta(subject)
+    blastp(querypath, subjectpath, flags)
+end
+
+function blastp(query::AminoAcidSequence, subject::AbstractString, flags=[]; db::Bool=false)
+    querypath = makefasta(query)
+    if db
+        blastp(querypath, subject, flags, db=true)
+    else
+        blastp(querypath, subject, flags)
+    end
+end
+
+function blastp{S <: AminoAcidSequence}(query::Vector{S}, subject::Vector{S}, flags=[])
+    querypath, subjectpath = makefasta(query), makefasta(subject)
+    blastp(querypath, subjectpath, flags)
+end
+
+"""
+`makefasta(sequence::BioSequence)`
+Create temporary fasta-formated file for blasting.
+"""
+function makefasta{T <: BioSequence}(sequence::T)
```

Comment:
There is no need to write a docstring that is not exported to users. Also it would be better to write the signature as `makefasta(sequence::BioSequence)`.